### PR TITLE
add toMatchSnapshot matcher

### PR DESCRIPTION
**Pull request by AnthonyQuy**
https://github.com/webdriverio/webdriverio/issues/9989

### CODE REVIEW

File: src/matchers/element/toMatchSnapshot.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+export function toMatchSnapshot(received: WdioElementMaybePromise, options: ExpectWebdriverIO.CommandOptions = {}) {
+    this.expectation = this.expectation || 'matchSnapshot'
+    return executeCommandBe.call(this, received, fnToMatchSnapShot, options)
+}
+
+async function fnToMatchSnapShot(element: WebdriverIO.Element, options?: ExpectWebdriverIO.CommandOptions): Promise<boolean> {
+    console.log(options);
+    try {
+        getSnapshotClient().assert({
+            received: element,
```
Comment (christian-bromann) [inline src/matchers/element/toMatchSnapshot.ts:28]: This would snapshot the `WebdriverIO.Element` rather than its HTML code right?

Comment (AnthonyQuy) [reply src/matchers/element/toMatchSnapshot.ts:28]: Actually it can be any object. Here I'm using as WebdriverIO.Element

Comment (christian-bromann) [reply src/matchers/element/toMatchSnapshot.ts:28]: I don't think snapshotting the `WebdriverIO.Element` is much useful, we should snapshot its HTML content.

Comment (AnthonyQuy) [reply src/matchers/element/toMatchSnapshot.ts:28]: I'm about to use it this way:

await expect({a:"a"}).toMatchSnapshot()
await expect($('#flash')).toMatchSnapshot()

so depending on what is the received Type we can handle it diff way

Comment (christian-bromann) [reply src/matchers/element/toMatchSnapshot.ts:28]: How about we keep `await expect({ a: 123 }).toMatchSnapshot()` to snapshot the object as in other frameworks and create a `await expect($('#flash')).toMatchHTMLSnapshot()` where we call `getHTML` from the object and snapshot that. Wdyt?

I prefer to keep things more implicit than hiding some *magic* beheind it.

**Pull request closed without merging**